tests/bluetooth/audio/bap_broadcast_source/src/main.c

```diff
@@ -219,3 +219,235 @@ ZTEST_F(bap_broadcast_source_test_suite, test_broadcast_source_create_start_send
+}
+
+ZTEST_F(bap_broadcast_source_test_suite,
+	test_broadcast_source_create_inval_subgroup_params_stream_params_null)
+{
+	struct bt_bap_broadcast_source_create_param *create_param = fixture->create_param;
+	struct bt_bap_broadcast_source_subgroup_param *subgroup_params = &create_param->params[0];
+	struct bt_bap_broadcast_source_stream_param *stream_params = &subgroup_params->params[0];
+	int err;
+
+	subgroup_params->params = NULL;
+	err = bt_bap_broadcast_source_create(create_param, &fixture->source);
+	/* Restore the params for the cleanup after function */
+	subgroup_params->params = stream_params;
+	zassert_not_equal(0, err, "Did not fail with NULL stream params");
+}
+
+ZTEST_F(bap_broadcast_source_test_suite,
+	test_broadcast_source_create_inval_subgroup_params_codec_cfg_null)
+{
+	struct bt_bap_broadcast_source_create_param *create_param = fixture->create_param;
+	struct bt_bap_broadcast_source_subgroup_param *subgroup_params = &create_param->params[0];
+	struct bt_audio_codec_cfg *codec_cfg = subgroup_params->codec_cfg;
+	int err;
+
+	subgroup_params->codec_cfg = NULL;
+	err = bt_bap_broadcast_source_create(create_param, &fixture->source);
+	/* Restore the params for the cleanup after function */
+	subgroup_params->codec_cfg = codec_cfg;
+	zassert_not_equal(0, err, "Did not fail with NULL codec_cfg");
+}
+
+ZTEST_F(bap_broadcast_source_test_suite,
+	test_broadcast_source_create_inval_subgroup_params_codec_cfg_data_len)
+{
+	struct bt_bap_broadcast_source_create_param *create_param = fixture->create_param;
+	struct bt_bap_broadcast_source_subgroup_param *subgroup_params = &create_param->params[0];
+	struct bt_audio_codec_cfg *codec_cfg = subgroup_params->codec_cfg;
+	int err;
+
+	codec_cfg->data_len = CONFIG_BT_AUDIO_CODEC_CFG_MAX_DATA_SIZE + 1;
+	err = bt_bap_broadcast_source_create(create_param, &fixture->source);
+	zassert_not_equal(0, err, "Did not fail with codec_cfg->data_len %zu", codec_cfg->data_len);
+}
+
+ZTEST_F(bap_broadcast_source_test_suite,
+	test_broadcast_source_create_inval_subgroup_params_codec_cfg_meta_len)
+{
+	struct bt_bap_broadcast_source_create_param *create_param = fixture->create_param;
+	struct bt_bap_broadcast_source_subgroup_param *subgroup_params = &create_param->params[0];
+	struct bt_audio_codec_cfg *codec_cfg = subgroup_params->codec_cfg;
+	int err;
+
+	codec_cfg->meta_len = CONFIG_BT_AUDIO_CODEC_CFG_MAX_METADATA_SIZE + 1;
+	err = bt_bap_broadcast_source_create(create_param, &fixture->source);
+	zassert_not_equal(0, err, "Did not fail with codec_cfg->meta_len %zu", codec_cfg->meta_len);
+}
+
+ZTEST_F(bap_broadcast_source_test_suite,
+	test_broadcast_source_create_inval_subgroup_params_codec_cfg_cid)
+{
+	struct bt_bap_broadcast_source_create_param *create_param = fixture->create_param;
+	struct bt_bap_broadcast_source_subgroup_param *subgroup_params = &create_param->params[0];
+	struct bt_audio_codec_cfg *codec_cfg = subgroup_params->codec_cfg;
+	int err;
+
+	codec_cfg->id = 0x06;
```

Comment:
Maybe use BT_HCI_CODING_FORMAT_LC3 from hci_types (PR: https://github.com/zephyrproject-rtos/zephyr/pull/62525 )

Reply:
Will update after one of the 2 PRs have been merged :)

Reply:
Updated